lib/tbot/index.ts

```diff
@@ -0,0 +1,83 @@
+import * as fs from 'fs/promises';
+import * as path from 'path';
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as yaml from 'yaml';
+
+import * as io from '../io';
+
+export interface SharedInputs {
+  proxy: string;
+  token: string;
+  debug: boolean;
+}
+
+export function getSharedInputs(): SharedInputs {
+  const proxy = core.getInput('proxy', { required: true });
+  const token = core.getInput('token', { required: true });
+
+  return {
+    proxy,
+    token,
+    debug: core.isDebug(),
+  };
+}
+
+// See https://github.com/gravitational/teleport/blob/master/lib/tbot/config/config.go#L206
+// For configuration references
+export interface ConfigurationV1Destination {
+  directory: {
+    path: string;
+  };
+  roles: Array<string>;
+  kubernetes_cluster?: string;
+}
+export interface ConfigurationV1 {
```

Comment:
Only one real nitpick here, it might make sense to expose `certificate_ttl` here if users have actions that take a longer time to run. It's never a great situation to be in, but I'm sure someone will need it eventually.

Reply:
31215df